File: demo/java/src/main/java/info/chenliang/tank/Server2ClientStub.java
package info.chenliang.tank;
import java.io.*;
import info.chenliang.talky.*;
public abstract class Server2ClientStub {
protected abstract DataOutputStream begin();
protected abstract void end();
public void onHit(byte attacker,short damage) throws Exception{
DataOutputStream dos = begin();
dos.write(0);
dos.writeByte(attacker);
dos.writeShort(damage);
end();
}
public void onSync(Player[] players) throws Exception{
DataOutputStream dos = begin();
dos.write(1);
SerializationUtil.writeVariableLength(dos, players.length);
for(int i=0; i < players.length;i++){
players[i].serialize(dos);
}
end();
}
}

File: demo/java/src/main/java/info/chenliang/tank/Server2ClientStub.cs
namespace info.chenliang.tank{
using System;
using System.IO;
using MiscUtil;
using MiscUtil.IO;
using MiscUtil.Conversion;
using Talky;
public abstract class Server2ClientStub {
protected abstract MemoryEndianBinaryWriter Begin();
protected abstract void End();
public void onHit(sbyte attacker,Int16 damage){
MemoryEndianBinaryWriter writer = Begin();
writer.WriteByte(0);
writer.WriteInt8(attacker);
writer.WriteInt16(damage);
End();
}
public void onSync(Player[] players){
MemoryEndianBinaryWriter writer = Begin();
writer.WriteByte(1);
SerializationUtil.WriteVariableLength(writer, players.Length);
for(int i=0; i < players.Length;i++){
players[i].Serialize(writer);
}
End();
}
}
}
